libraries/standard/http/src/http_client.c

```diff
@@ -16,22 +48,162 @@ HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
     return HTTP_NOT_SUPPORTED;
 }
 
+/*-----------------------------------------------------------*/
+
 HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
                                         int32_t rangeStart,
                                         int32_t rangeEnd )
 {
     return HTTP_NOT_SUPPORTED;
 }
 
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _sendHttpHeaders( const HTTPTransportInterface_t * pTransport,
+                                      const HTTPRequestHeaders_t * pRequestHeaders )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    int32_t transportStatus = 0;
+
+    /* Send the HTTP headers over the network. */
+    transportStatus = pTransport->send( pTransport->pContext,
+                                        pRequestHeaders->pBuffer,
+                                        pRequestHeaders->headersLen );
+
+    if( transportStatus < 0 )
+    {
+        IotLogErrorWithArgs( "Error in sending the HTTP headers over the transport "
+                             "interface: Transport status %d.",
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else if( transportStatus != pRequestHeaders->headersLen )
+    {
+        IotLogErrorWithArgs( "Failure in sending HTTP headers: Transport layer "
+                             "sent less than the required bytes: Required Bytes=%d"
+                             ", Sent Bytes=%d.",
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else
+    {
+        /* Empty else MISRA 15.7 */
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _sendHttpBody( const HTTPTransportInterface_t * pTransport,
+                                   const uint8_t * pRequestBodyBuf,
+                                   size_t reqBodyBufLen )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    int32_t transportStatus = 0;
+
+    /* Send the HTTP body over the network. */
+    if( pRequestBodyBuf != NULL )
+    {
+        transportStatus = pTransport->send( pTransport->pContext,
+                                            pRequestBodyBuf,
+                                            reqBodyBufLen );
+
+        if( transportStatus < 0 )
+        {
+            IotLogErrorWithArgs( "Error in sending the HTTP body over the "
```

Comment:
This log message will probably not be very helpful, since `transportStatus` will have a value like `-119` or something.

Reply:
Then it's also not very helpful to pass it through back to the application well?